libraries/rush-lib/src/cli/actions/BaseInstallAction.ts

```diff
@@ -244,9 +244,12 @@ export abstract class BaseInstallAction extends BaseRushAction {
         for (const selectedSubspace of selectedSubspaces) {
           installManagerOptions.subspace = selectedSubspace;
           if (selectedSubspace.getPnpmOptions()?.alwaysFullInstall) {
-            installManagerOptions.filteredProjects = [];
+            installManagerOptions.filteredProjects = undefined;
           } else {
-            installManagerOptions.filteredProjects = filteredProjectsForSubspace.get(selectedSubspace) || [];
+            const filteredProjects: Set<RushConfigurationProject> | undefined =
+              filteredProjectsForSubspace.get(selectedSubspace);
+            installManagerOptions.filteredProjects =
+              filteredProjects?.size === selectedSubspace.getProjects().length ? undefined : filteredProjects;
```

Comment:
Add a `//` comment here explaining that, if `filteredProjects` is the entire set of projects, then specify `undefined` which disables filtering entirely, producing a full install. This is not obvious to a casual reader of the code.